lib/assets/javascripts/turbolinks.js.coffee

```diff
@@ -203,8 +202,19 @@ onNodeRemoved = (node) ->
     jQuery(node).remove()
   triggerEvent(EVENTS.AFTER_REMOVE, node)
 
-executeScriptTags = (selector) ->
-  scripts = document.body.querySelectorAll(selector)
+getScriptsToRun = (runScripts) ->
+  selector = if runScripts is false then 'script[data-turbolinks-eval="always"]' else 'script:not([data-turbolinks-eval="false"])'
+  scripts = document.querySelectorAll(selector)
+  script for script in scripts when script.type in ['', 'text/javascript'] and not withinPermanent(script)
+
+withinPermanent = (element) ->
+  while element?
+    return true if element.hasAttribute && element.hasAttribute('data-turbolinks-permanent')
```

Comment:
The `document` node doesn't implement `hasAttribute`. We could check `nodeType` instead if you think checking for `hasAttribute` is too opaque.